receive_stream.go

```diff
@@ -0,0 +1,284 @@
+package quic
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/internal/flowcontrol"
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/utils"
+	"github.com/lucas-clemente/quic-go/internal/wire"
+)
+
+type receiveStream struct {
+	mutex sync.Mutex
+
+	streamID protocol.StreamID
```

Comment:
Would it make sense to encapsulate the mutex and streamID into a baseStream?

Reply:
If you  share the `baseStream` between the `sendStream` and the `receiveStream`, you'd only have a single mutex instead of two. Two is obviously better here.